op-node/rollup/engine/events.go

```diff
@@ -323,6 +325,29 @@ func (d *EngDeriver) OnEvent(ev event.Event) bool {
 			}
 		} else {
 			d.log.Info("successfully processed payload", "ref", ref, "txs", len(x.Envelope.ExecutionPayload.Transactions))
+			payload := x.Envelope.ExecutionPayload
+			if payload != nil {
+				latestBlockTimeStamp := uint64(payload.Timestamp)
+
+				currentTime := uint64(time.Now().Unix())
+
+				if latestBlockTimeStamp <= currentTime {
+					timeDiff := currentTime - latestBlockTimeStamp
+
+					if timeDiff < blockTimeThreshold {
+						// Node is healthy
+						d.log.Info("Node is healthy, time difference within threshold",
+							"time_diff", timeDiff, "threshold", blockTimeThreshold)
+					} else {
+						// Node is stale
+						d.log.Warn("Node is stale, time difference exceeds threshold",
+							"time_diff", timeDiff, "threshold", blockTimeThreshold)
```

Comment:
I think we should use metrics here (specifically a counter) instead of using logs.